bin/bump_version.nim

```diff
@@ -0,0 +1,267 @@
+#!/usr/bin/env -S nim r --verbosity:0 --skipParentCfg:on
+
+## Running this file will:
+## 1. Switch to the `main` branch, if there are no uncommitted changes (exiting
+##    otherwise).
+## 2. Pull the upstream `main` branch from github.com/exercism/configlet.
+## 3. Check that the local `main` is not ahead of upstream.
+## 4. Create and switch to a new git branch named like `release-x.y.z`.
+## 5. Increment the version in `configlet.version` if it exists, otherwise in
+##    the Nimble file (the former is required to support pre-release version
+##    strings like `4.0.0-alpha.36`).
+## 6. Commit the change.
+## 7. If `gh` is installed, prompt the user to push to GitHub and create the PR.
+## 8. Switch back to the `main` branch.
+import std/[os, osproc, strformat, strscans, strutils]
+
+type
+  BumpError = object of CatchableError
+  BumpQuit = object of CatchableError
+
+proc error(msg: string) =
+  ## Raises a `BumpError`.
+  raise newException(BumpError, msg)
+
+proc exit0(msg: string) =
+  ## Raises a `BumpQuit`.
+  raise newException(BumpQuit, msg)
+
+type
+  Command = enum
```

Comment:
Yeah, it's an interesting readability benefit to have a list of commands that the `exec`  procs are able to run.

An implementation detail here: for brevity, I wrote the `exec` proc using `execCmdEx`, which has a backward compatibility pitfall of calling `startProcess` with the `poEvalCmd` option. 

See:
- [osproc.execCmdEx](https://github.com/nim-lang/Nim/blob/731eabc9309997775c8be41f3e5eb5512460aad0/lib/pure/osproc.nim#L1612-L1613)
- [osproc.startProcess](https://github.com/nim-lang/Nim/blob/731eabc9309997775c8be41f3e5eb5512460aad0/lib/pure/osproc.nim#L137-L143)
- [osproc.ProcessOption](https://github.com/nim-lang/Nim/blob/731eabc9309997775c8be41f3e5eb5512460aad0/lib/pure/osproc.nim#L41-L42)

So for the `exec` proc to only be able to run the enumerated commands, the `quoteShell` on line 50 is essential:
https://github.com/exercism/configlet/blob/d603a1d1517ec729ad4e35ffe733b305bde6ca85/bin/bump_version.nim#L45-L51

Because if line 50 was instead just `cmd.add arg`, then the output of

```Nim
echo exec(GitRevParse, ["main", "&&", "rm", "yourhomedirectory"])[0]
```

is

```
6364d9990828323b1e1ab4a0255924ab844dea7d
rm: cannot remove 'yourhomedirectory': No such file or directory
```

rather than

```
fatal: ambiguous argument '&&': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
6364d9990828323b1e1ab4a0255924ab844dea7d
&&
```

In my own code, I've often use a tweaked `execCmdEx` that avoids `poEvalCmd`.

I'll add a command for the `quoteShell` detail.